File: app/src/main/java/com/example/socialapp/ui/home/HomeRecyclerViewAdapter.java
package com.example.socialapp.ui.home;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.socialapp.R;
import com.example.socialapp.models.Post;

import java.util.List;

public class HomeRecyclerViewAdapter extends RecyclerView.Adapter<HomeRecyclerViewAdapter.PostViewHolder>{
    List<Post> posts;

    public HomeRecyclerViewAdapter(List<Post> posts) {

        this.posts = posts;
    }

    @NonNull
    @Override
    public PostViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_post,parent,false);
        return new PostViewHolder(view);
    }
    @Override
    public void onBindViewHolder(@NonNull PostViewHolder holder, int position) {
        holder.name.setText(posts.get(position).getName());
    }
    @Override
    public int getItemCount() {
        return posts.size();
    }

    class PostViewHolder extends RecyclerView.ViewHolder {

       TextView name;

        public PostViewHolder(@NonNull View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.tv1);
        }
    }
    public  void update(List<Post> posts){
        if(!posts.isEmpty()){
            this.posts.clear();
        }
        this.posts.addAll(posts);
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/example/socialapp/ui/home/HomeRecyclerViewAdapter.kt
package com.example.socialapp.ui.home
import com.example.socialapp.models.Post
import androidx.recyclerview.widget.RecyclerView
import com.example.socialapp.ui.home.HomeRecyclerViewAdapter.PostViewHolder
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.example.socialapp.R
import android.widget.TextView

class HomeRecyclerViewAdapter(var posts: MutableList<Post>) :
    RecyclerView.Adapter<PostViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PostViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_post, parent, false)
        return PostViewHolder(view)
    }
    override fun onBindViewHolder(holder: PostViewHolder, position: Int) {
        holder.name.text = posts[position].name
    }
    override fun getItemCount(): Int {
        return posts.size
    }
    inner class PostViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var name: TextView = itemView.findViewById(R.id.tv1)
    }
    fun update(posts: List<Post>) {
        if (!posts.isEmpty()) {
            this.posts.clear()
        }
        this.posts.addAll(posts)
        notifyDataSetChanged()
    }
}